catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java

```diff
@@ -399,5 +404,28 @@ private void updateTeams(User origUser, User updatedUser) throws IOException {
       List<EntityReference> deleted = new ArrayList<>();
       recordListChange("teams", origTeams, updatedTeams, added, deleted, EntityUtil.entityReferenceMatch);
     }
+
+    private void updateAuthenticationMechanism(User origUser, User updatedUser) throws IOException {
+      AuthenticationMechanism origAuthMechanism = origUser.getAuthenticationMechanism();
+      AuthenticationMechanism updatedAuthMechanism = updatedUser.getAuthenticationMechanism();
+      if (origAuthMechanism == null && updatedAuthMechanism != null) {
+        recordChange(
+            "authenticationMechanism", origUser.getAuthenticationMechanism(), updatedUser.getAuthenticationMechanism());
+      } else if (origAuthMechanism != null
+          && updatedAuthMechanism != null
+          && origAuthMechanism.getConfig() != null
+          && updatedAuthMechanism.getConfig() != null) {
+        JWTAuthMechanism origJwtAuthMechanism =
```

Comment:
We have && in the previous line no :)

Reply:
the first check is to set when the existing authMech is empty and the second check is to check if there is any change between the existing authMech and the new authMech. 
I'll re-read the code again tomorrow, might be missing something.